webui/react/src/pages/Authentication.tsx

```diff
@@ -0,0 +1,156 @@
+import { Button, Form, Input } from 'antd';
+import { message } from 'antd';
+import axios from 'axios';
+import queryString from 'query-string';
+import React, { useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router';
+import { Redirect } from 'react-router-dom';
```

Comment:
It might be better to use the `useHistory` hook from `react-router-dom` instead of `react-router` since we are creating our own history object in `src/index.tsx` and passing that into the `Router` component from `react-router-dom`.